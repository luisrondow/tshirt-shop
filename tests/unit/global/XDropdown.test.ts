import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

import { XDropdown } from '#components'

describe('XDropdown component', () => {
  it('should render correctly', () => {
    const wrapper = mount(XDropdown, {
      props: {
        items: [
          { value: 'test', label: 'Test' },
          { value: 'test-2', label: 'Test 2' },
        ],
        modelValue: 'test',
      },
    })

    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should open dropdown', async () => {
    const wrapper = mount(XDropdown, {
      props: {
        items: [
          { value: 'test', label: 'Test' },
          { value: 'test-2', label: 'Test 2' },
        ],
        modelValue: 'test',
      },
    })

    await wrapper.find('[data-testid="dropdown-select"]').trigger('click')

    expect(wrapper.find('[data-testid="dropdown-options"]').isVisible()).toBeTruthy()
    expect(wrapper.find('[data-testid="dropdown-options"]').text()).toContain('Test')
    expect(wrapper.find('[data-testid="dropdown-options"]').text()).toContain('Test 2')
    expect(wrapper.findAll('button').length).toBe(2)
  })

  it('should select option', async () => {
    const wrapper = mount(XDropdown, {
      props: {
        items: [
          { value: 'test', label: 'Test' },
          { value: 'test-2', label: 'Test 2' },
        ],
        modelValue: 'test',
      },
    })

    await wrapper.find('[data-testid="dropdown-select"]').trigger('click')
    await wrapper.findAll('button')[1].trigger('click')

    expect(wrapper.emitted('update:modelValue')).toBeTruthy()
    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual(['test-2'])
  })
})
