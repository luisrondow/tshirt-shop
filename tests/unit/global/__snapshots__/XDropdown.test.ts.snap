// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`XDropdown component > should render correctly 1`] = `
<div
  class="dropdown relative"
>
  <div
    class="flex h-10 min-w-24 cursor-pointer items-center rounded border border-gray-200 bg-white p-3"
    data-testid="dropdown-select"
  >
    <span>
      test
    </span>
    <svg
      class="ml-auto h-4 w-4"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 9l-7 7-7-7"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </div>
  <!--v-if-->
</div>
`;
